documentprocessorfactory.h

```diff
@@ -0,0 +1,61 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 N7 Mobile sp. z o. o.
+** Contact: http://n7mobile.pl/Space
+**
+** This file is part of ASN.1/ACN Plugin for QtCreator.
+**
+** Plugin was developed under a programme and funded by
+** European Space Agency.
+**
+** This Plugin is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** This Plugin is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#pragma once
+
+#include "documentprocessor.h"
+#include "documentprocessorstub.h"
+#include "asn1sccdocumentprocessor.h"
+
+namespace Asn1Acn {
+namespace Internal {
+
+class DocumentProcessorFactory
+{
+public:
+    enum class WorkMode {
+        Test,
+        Real
+    };
+
+    DocumentProcessorFactory(WorkMode mode = WorkMode::Real) : m_mode(mode) {}
+
+    DocumentProcessor *create(const QString &projectName) const {
+        switch(m_mode) {
+            case WorkMode::Test:
```

Comment:
that's ugly - it makes tests code influence production code - tests could and should influence design of production code, but production code should not be "aware" of tests

Reply:
This is to test ProjectContentHandler class. The class uses instances of DocumentProcessor. It used to create them internally. Overall approach to make the classes be testable is to pass other objects from the outside. In this case it was not possible to be done straightforward, as multiple instances of DocumentProcessor are needed, and it was ProjectContentHandler is responsible to create all of them. This is why I came with the idea of this factory, but I will change it.